vlog4j-core/src/main/java/org/semanticweb/vlog4j/core/reasoner/util/ModelToVLogConverter.java

```diff
@@ -22,74 +24,135 @@
+		Validate.notNull(predicate);
+		final String vLogPredicate = predicate.getName() + PREDICATE_ARITY_SUFFIX_SEPARATOR + predicate.getArity();
+		return vLogPredicate;
+	}
+
 	public static karmaresearch.vlog.Atom toVLogAtom(final Atom atom) {
-		// TODO treat null case: throw exception or return null?
-		final karmaresearch.vlog.Term[] terms = toVLogTermArray(atom.getTerms());
-		// FIXME: Append arity at the end of the predicate name (punning)
-		// FIXME: should we generate predicate names by appending the terms arity to the
-		// given name?
-		final karmaresearch.vlog.Atom vLogAtom = new karmaresearch.vlog.Atom(atom.getPredicate().getName(), terms);
+		Validate.notNull(atom);
+		final karmaresearch.vlog.Term[] vlogTerms = toVLogTermArray(atom.getTerms());
+		final String vLogPredicate = toVlogPredicate(atom.getPredicate());
+		final karmaresearch.vlog.Atom vLogAtom = new karmaresearch.vlog.Atom(vLogPredicate, vlogTerms);
 		return vLogAtom;
 	}
 
-	public static karmaresearch.vlog.Atom[] toVLogAtomArray(final Conjunction atoms) {
-		// TODO treat null case: throw exception or return null?
-		final karmaresearch.vlog.Atom[] vlogAtoms = new karmaresearch.vlog.Atom[atoms.getAtoms().size()];
+	private static karmaresearch.vlog.Atom[] toVLogAtomArray(final Conjunction conjunction) {
+		final karmaresearch.vlog.Atom[] vlogAtoms = new karmaresearch.vlog.Atom[conjunction.getAtoms().size()];
 		int i = 0;
-		for (final Atom atom : atoms) {
+		for (final Atom atom : conjunction.getAtoms()) {
 			vlogAtoms[i] = toVLogAtom(atom);
 			i++;
 		}
 		return vlogAtoms;
 	}
 
-	public static karmaresearch.vlog.Rule toVLogRule(final Rule rule) {
-		// TODO treat null case: throw exception or return null?
+	private static karmaresearch.vlog.Rule toVLogRule(final Rule rule) {
+		Validate.notNull(rule);
 		final karmaresearch.vlog.Atom[] vlogHead = toVLogAtomArray(rule.getHead());
 		final karmaresearch.vlog.Atom[] vlogBody = toVLogAtomArray(rule.getBody());
 		return new karmaresearch.vlog.Rule(vlogHead, vlogBody);
 	}
 
 	public static karmaresearch.vlog.Rule[] toVLogRuleArray(final List<Rule> rules) {
-		// TODO treat null case: throw exception or return null?
+		Validate.noNullElements(rules);
 		final karmaresearch.vlog.Rule[] vlogRules = new karmaresearch.vlog.Rule[rules.size()];
-		for (int i = 0; i < rules.size(); i++) {
-			vlogRules[i] = toVLogRule(rules.get(i));
+		int i = 0;
+		for (final Rule rule : rules) {
+			vlogRules[i] = toVLogRule(rule);
+			i++;
 		}
 		return vlogRules;
 	}
+
+	public static karmaresearch.vlog.VLog.RuleRewriteStrategy toVLogRuleRewriteStrategy(
+			final RuleRewriteStrategy ruleRewriteStrategy) {
+		Validate.notNull(ruleRewriteStrategy);
+		final karmaresearch.vlog.VLog.RuleRewriteStrategy vLogRuleRewriteStrategy;
+		switch (ruleRewriteStrategy) {
+		case SPLIT_HEAD_PIECES_AGGRESIVE:
+			vLogRuleRewriteStrategy = karmaresearch.vlog.VLog.RuleRewriteStrategy.AGGRESSIVE;

```

Comment:
you can just inline the return here and safe the breaks

Reply:
done